test/test_fcntl.rs

```diff
@@ -561,3 +561,27 @@ mod test_posix_fallocate {
         }
     }
 }
+
+#[cfg(any(
+    target_os = "dragonfly",
+    target_os = "netbsd",
+    target_os = "macos",
+    target_os = "ios"
+))]
+mod test_apple_netbsd {
+    use nix::fcntl::*;
+    use std::os::unix::io::AsRawFd;
+    use std::path::PathBuf;
+    use tempfile::NamedTempFile;
+
+    #[test]
+    fn test_path() {
+        let tmp = NamedTempFile::new().unwrap();
+        let fd = tmp.as_raw_fd();
+        let mut path = PathBuf::new();
+        let res =
+            fcntl(fd, FcntlArg::F_GETPATH(&mut path)).expect("get path failed");
+        assert_ne!(res, -1);
+        assert_eq!(path.as_path().file_name(), tmp.path().file_name());
```

Comment:
Is that assertion failure the expected behavior? I think we need to figure out why they are giving different paths

Reply:
I wonder why tempnamedfile (another macos oddity) does not store the fullpath, even tough in practice they are the same paths.

Reply:
From this [answer's comment](https://apple.stackexchange.com/a/131080/394687), it seems that `/var` is a symlink to `/private/var`, so if `fcntl()` follows symlink, then the behavior is reasonable.

It would be great if we document this behavior in the test, explaining a bit about why we are only comparing the file names

Reply:
```
   Compiling tempfile v3.8.0
error: unused doc comment
   --> test/test_fcntl.rs:585:9
    |
585 | /         /// We purposely check the filenames, since they are supposed to be unique.
586 | |         /// The left operand tend to have `/private/var` whereas the right does not contains
587 | |         /// the `/private` part even tough in practice they lead to the same file.
    | |_________-------------------------------------------------------------------------^
    |           |
    |           rustdoc does not generate documentation for macro invocations
    |
    = note: `-D unused-doc-comments` implied by `-D warnings`
    = help: to document an item produced by a macro, the macro must produce the documentation as part of its expansion

error: could not compile `nix` due to previous error
warning: build failed, waiting for other jobs to finish...
```

Comment starts with `///` is a doc comment, we should use code comment  (`// comment`) instead.

And, it is kinda necessary to mention that `/var` is a symlink to `/private/var` and `fcntl()` follows symlinks, then we can say that `in practice they lead to the same file`